extensions/podman/src/lifecycle-context.ts

```diff
@@ -0,0 +1,77 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { LifecycleContext, Logger } from '@podman-desktop/api';
+
+// This file is the same as in the main repo (plugin/lifecycle-context.ts)
+// but added to the extension in order to implement logging for lifecycle context.
```

Comment:
Needs to be added / copy-and-pasted over so we can use the logging functionality to call the extension + restart the machines. Which seems to be the most cleanest / efficient way rather than using `extension.ts` functions.